src/xrpld/app/tx/detail/Escrow.cpp

```diff
@@ -448,13 +533,15 @@ EscrowFinish::doApply()
             return tecCRYPTOCONDITION_ERROR;
     }
 
-    // NOTE: Escrow payments cannot be used to fund accounts.
     AccountID const destID = (*slep)[sfDestination];
     auto const sled = ctx_.view().peek(keylet::account(destID));
-    if (!sled)
-        return tecNO_DST;
 
-    if (ctx_.view().rules().enabled(featureDepositAuth))
+    if (ctx_.tx.isFieldPresent(sfCredentialIDs))
+    {
+        if (Credentials::removeExpired(view(), ctx_.tx, j_))
```

Comment:
Equivalent check (perhaps with `Credentials::checkExpired`) could be done in `preclaim` ; it would be slightly cheaper as there would be no need to lookup credentials second time.

Reply:
It should be done here because this code change ledger (delete expired credentials) despite the transaction will fail with tecExpired. It is some sort of garbage collector

Reply:
Ah I see. So basically we need to `Transactor::doApply` to execute the transaction in order for the new ledger with this transaction to be produced, even if it's only for a side effect (i.e. removing expired credentials). Makes sense !